core/services/blockhashstore/feeder.go

```diff
@@ -63,6 +82,34 @@ type Feeder struct {
 	errsLock      sync.Mutex
 }
 
+func (f *Feeder) subscribeHeadBroadcaster() {
+	f.headBroadcaster.Subscribe(f)
+}
+
+// OnNewLongestChain implements HeadTrackable
+// note that the ctx has a timeout of 2 seconds
+// the below code cannot exceed that else we start missing
+// heartbeat bh's.
+func (f *Feeder) OnNewLongestChain(ctx context.Context, head *evmtypes.Head) {
+	if diff := head.Timestamp.Sub(f.storeEarliest); diff.Seconds() >= float64(f.heartbeatPeriodTime) {
+		f.lggr.Infow("storing heartbeat blockhash",
+			"latestHead", head.Number,
+			"heartbeatPeriodTime (in seconds)", f.heartbeatPeriodTime,
+			"storeEarliest", f.storeEarliest,
+		)
+		if err := f.bhs.Store(ctx, uint64(head.Number)); err != nil {
```

Comment:
maybe I'm being overly cautious here but some minor optimizations worth considering: 
- `storeEarliest` instead of 'store' so that we don't rely on head tracker accuracy (e.g. head tracker can fall behind in case of bad RPC) 
- `time.Now()` or similar instead of head.Timestamp because most blockchains don't impose strict limits on `head.Timestamp` (i.e. time can be arbitrary)

Reply:
Used Go timer and storeEarliest instead of headBroadcaster